### PR TITLE
Fix 16506 Hide_db and Only_db does not work

### LINKED ISSUE #16506: hide_db seems to not work
The hide_db configuration directive seems to have no effect.

I have set `$cfg['Servers'][$i]['hide_db'] = 'sakila';` and yet I still see the sakila database listed in the navigation pane. I've also tried ``$cfg['Servers'][$i]['hide_db'] = '^(sakila|gis)$';`` again with no luck.

I've tested with MariaDB 10.3 and MySQL 8.0, and phpMyAdmin 4.9.1 and 5.1.0-dev from master.

### PR DESCRIPTION
**Pull request by shucon**
Signed-off-by: Saksham Gupta <shucon01@gmail.com>

### Description

Fixes #16506

Before submitting pull request, please review the following checklist:

- [ ] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [ ] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

File: libraries/classes/UserPreferences.php [hunk 1]
```diff
@@ -118,6 +118,11 @@ public function save(array $config_array)
         $cfgRelation = $this->relation->getRelationsParam();
         $server = $GLOBALS['server'] ?? $GLOBALS['cfg']['ServerDefault'];
         $cache_key = 'server_' . $server;
+        foreach ($config_array as $key => $value) {
+            if ($value == "") {
```
Comment (shucon) [inline libraries/classes/UserPreferences.php:122]: We could also add a condition where `$key` can only be `hide_db` or `only_db` but I think it's not required.

**Review by williamdes — CHANGES_REQUESTED** [review]
This will maybe have side effects on users wanting to remove the value set by the config

**Review by MauricioFauth — CHANGES_REQUESTED** [review]
This is not the correct solution. The issue is that after changing a setting through the Settings page, the value in the `config.inc.php` is ignored.

**Comment by shucon** [comment]
What do you propose? What should be the desired behaviour when there's a value in both?

**Pull request closed without merging**